src/routes/blog/rss.xml/+server.ts

```diff
@@ -3,23 +3,40 @@ import { posts } from '../content';
 
 export const prerender = true;
 
+function encodeTitle(str: string): string {
+    return str.replace(/&/g, '&amp;')
+        .replace(/</g, '&lt;')
+        .replace(/>/g, '&gt;')
+        .replace(/"/g, '&quot;')
+        .replace(/'/g, '&#39;');
+}
+
+function encodeUrl(href: string): string {
+    return encodeURI(`https://appwrite.io${href}`)
+}
+
 export const GET: RequestHandler = () => {
     const feed = `<?xml version="1.0" encoding="UTF-8" ?>
 <rss version="2.0" xmlns:atom="http://www.w3.org/2005/Atom">
 <channel>
     <title>Appwrite</title>
     <link>https://appwrite.io</link>
+    <atom:link href="https://appwrite.io/blog/rss.xml" rel="self" type="application/rss+xml" />
     <description>Appwrite is an open-source platform for building applications at any scale, using your preferred programming languages and tools.</description>
     ${posts.map((post) => `<item>
-        <title>${post.title}</title>
+        <title>${encodeTitle(post.title)}</title>
         <pubDate>${post.date.toUTCString()}</pubDate>
-        <link>https://appwrite.io${post.href}</link>
-        <guid>https://appwrite.io${post.href}</guid>
+        <link>${encodeUrl(post.href)}</link>
+        <guid>${encodeUrl(post.href)}</guid>
         <description>${post.description}</description>
```

Comment:
I tihnk descrition needs to be encoded too - I am not sure about the urls though

Reply:
I could add encoding to description, kinda makes sense.
One of the urls had a space and that would really break the xml, so I added encoding to urls too.